packages/components/src/components/post-icon/post-icon.tsx

```diff
@@ -1,37 +1,95 @@
 import { Component, Host, h, Prop, State } from '@stencil/core';
 
+/**
+ * @class PostIcon - representing a stencil component
+ * 
+ * @param {string} name - The name/id of the icon (e.g. 1000, 1001, ...).
+ * @param {string} base - The base path, where the icons are located (must be a public url).
+ * @param {boolean} flipH - When set to `true`, the icon will be flipped horizontally.
+ * @param {boolean} flipV - When set to `true`, the icon will be flipped vertically.
+ * @param {number} scale - The `number` for the css `scale` transformation.
+ * @param {number} rotate - The `number` of degree for the css `rotate` transformation.
+ * @param {boolean} animation - The name of the animation (`cylon`, `cylon-vertical`, `spin`, `spin-reverse`, `fade`, `throb`).
+ * 
+ * @exports {class} PostIcon - Which will be processed to a custom web-component on build.
+ * 
+ * @see: https://jsdoc.app/ and https://en.wikipedia.org/wiki/JSDoc
+ */
 @Component({
   tag: 'post-icon',
   styleUrl: 'post-icon.scss',
   shadow: true,
 })
 export class PostIcon {
-  @Prop() name: number;
+  @Prop() name: string;
   @Prop() base?: string;
+  @Prop() flipH?: boolean;
+  @Prop() flipV?: boolean;
+  @Prop() scale?: number;
+  @Prop() rotate?: number;
+  @Prop() animation?: string;
   @State() path: string;
+  @State() svgSource: string = '<svg viewBox="0 0 16 16"></svg>';
+  @State() svgOutput: string;
 
-  constructor() {
+  connectedCallback () {
     // Construct icon path from different possible sources
     let basePath: string;
     const metaBase = document.head.querySelector(
       'meta[name="design-system-settings"][data-post-icon-base]',
     );
-    if (metaBase) basePath = metaBase.getAttribute('data-post-icon-base');
-    if (this.base) basePath = this.base;
-    if (!basePath) basePath = `https://unpkg.com/@swisspost/design-system-icons/public/svg`;
-
+  
+    if (this.base) {
+      basePath = this.base;
+    } else if (metaBase) {
+      basePath = metaBase.getAttribute('data-post-icon-base');  
+    } else {
+      basePath = 'https://unpkg.com/@swisspost/design-system-icons/public/post-icons';
+    }
+  
     this.path = new URL(
       [...basePath.split('/'), `${this.name}.svg#icon`].join('/'),
       window.location.origin,
     ).toString();
+
+    this.svgSource = window.localStorage.getItem(`post-icon-${this.name}`) ?? this.svgSource;
+  }
+
+  componentWillLoad () {
+    fetch(this.path)
+      .then(response => response.text())
+      .then(svgSource => {
+        this.svgSource = svgSource;
+        window.localStorage.setItem(`post-icon-${this.name}`, svgSource);
+      })
+      .catch(error => {
+        console.log(error);
+      });
```

Comment:
This approach creates a problem if the response code is 200 but it's an error page:

![image](https://user-images.githubusercontent.com/1659006/214802685-eb6eafa1-dbed-4ee2-929a-e3a6dd22eee9.png)

In this case, the contents of the error page from netlify will be rendered to the page, making for some really strange stuff that can happen. We should:
 - Catch this case if possible (checking for SVG only response)
 - Consider using a fallback for errors, e.g. if it's a custom URL and it resolves to an error, fall back to loading from CDN

Reply:
Should work now. Make sure you clean your browser cache before you load the url ;)